Form/FormMapper.php

```diff
@@ -89,6 +89,11 @@ public function add($name, $type = null, array $options = array(), array $fieldD
 
         $group = $this->addFieldToCurrentGroup($label);
 
+        // Try to autodetect type
+        if ($name instanceof FormBuilderInterface && $type == null) {
```

Comment:
> We use yoda conditions and strict checking in many methods at the moment.

I know but if we can avoid these questionable practices with a dedicated function, why not? But `is_null()` is not that much better in terms of Yodaism… so, in this particular case, I'm not so sure what is best .